lib/node/http.js

```diff
@@ -432,6 +434,131 @@ class HTTP extends Server {
+      const name = valid.str('name');
+
+      if (!name || !rules.verifyName(name))
+        throw new Error('Invalid parameter.');
+
+      const network = this.network;
+      const height = this.chain.height;
+      const nameHash = rules.hashName(name);
+      const reserved = rules.isReserved(nameHash, height + 1, network);
+      const [start, week] = rules.getRollout(nameHash, network);
+      const ns = await this.chain.db.getNameState(nameHash);
+
+      let info = null;
+
+      if (ns) {
+        if (!ns.isExpired(height, network))
+          info = ns.getJSON(height, network);
+      }
+
+      return res.json(200, {
+        start: {
+          reserved: reserved,
+          week: week,
+          start: start
+        },
+        info
+      });
+    });
+
+    this.get('/resource/hash/:hash', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const hash = valid.bhash('hash');
+
+      if (!hash)
+        throw new Error('Invalid hash.');
+
+      const ns = await this.chain.db.getNameState(hash);
+      const height = this.chain.tip.height;
+      const network = this.network;
+
+      if (!ns)
+        return res.json(404);
+
+      return res.json(200, ns.getJSON(height,network));
+    });
+
+    this.get('/resource/name/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      if (!name || !rules.verifyName(name))
+        throw new Error('Invalid name.');
+
+      const nameHash = rules.hashName(name);
+      const ns = await this.chain.db.getNameState(nameHash);
+
+      if (!ns || ns.data.length === 0)
+        return null;
+
+      const resource = Resource.decode(ns.data);
+
+      return res.json(200, resource.getJSON(name));
+    });
+
+    this.get('/proof/name/:name', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      if (!name || !rules.verifyName(name))
+        throw new Error('Invalid name.');
+
+      const tip = this.chain.tip.hash;
+      const height = this.chain.tip.height;
+      const root = this.chain.tip.treeRoot;
+      const nameHash = rules.hashName(name);
+      const proof = await this.chain.db.prove(root, nameHash);
+
+      return res.json(200, {
+        hash: tip.toString('hex'),
+        height: height,
+        root: root.toString('hex'),
+        name: name,
+        key: nameHash.toString('hex'),
+        proof: proof.toJSON()
+      });
+    });
+
+    this.get('/proof/hash/:nameHash', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const nameHash = valid.bhash('nameHash');
+
+      if (!nameHash)
+        throw new Error('Invalid hash.');
+
+      const tip = this.chain.tip.hash;
+      const height = this.chain.tip.height;
+      const root = this.chain.tip.treeRoot;
+      const proof = await this.chain.db.prove(root, nameHash);
+
+      return res.json(200, {
```

Comment:
@tynes :eyes: We ok with this response shape? Mirrors the behavior of `getnameproof` but omits `.name`